src/app/qgsmaptooladdfeature.cpp

```diff
@@ -121,31 +121,45 @@ void QgsMapToolAddFeature::featureDigitized( const QgsFeature &feature )
     }
     if ( topologicalEditing )
     {
-      const QList<QgsPointLocator::Match> sm = snappingMatches();
-      for ( int i = 0; i < sm.size() ; ++i )
+      const QVector<QgsVectorLayer *> layers = vlayer->project()->layers<QgsVectorLayer *>();
+
+      for ( QgsVectorLayer *layer : layers )
```

Comment:
We should probably rely on the canvas' layers and not the project's layers:

```suggestion
      const QList<QgsMapLayer *> layers = canvas()->layers( true );

      for ( QgsMapLayer *layer : layers )
      {
        QgsVectorLayer *vectorLayer = qobject_cast<QgsVectorLayer *>( layer );
        if ( vectorLayer ...
```